### PR TITLE
If only one function is passed to composeK, composeK will return that…

### PR DESCRIPTION
**Pull request by niksosf**
Related to issue [#2380 ](https://github.com/ramda/ramda/issues/2380)
where if only one function is passed to `composeK`, `composeK` will try to pass an empty array to `compose` which will throw as it requires at least one argument.

### CODE REVIEW

File: source/composeK.js [hunk 1]
```diff
@@ -39,5 +38,9 @@ export default function composeK() {
   }
   var init = Array.prototype.slice.call(arguments);
   var last = init.pop();
-  return compose(compose.apply(this, map(chain, init)), last);
+  if (init.length === 0) {
+    last;
```
Comment (skatcat31) [inline source/composeK.js]: Should this have a return in front of it?

File: source/composeK.js [hunk 2]
```diff
@@ -39,5 +38,9 @@ export default function composeK() {
...
+  } else {
```
Comment (skatcat31) [inline source/composeK.js]: Assuming previous comment is a return, isn't the else here unneeded?

File: source/composeK.js (hunk 1 above)
Comment (niksosf) [reply source/composeK.js]: Definitely! Should I make a change to re-make a pull request?

File: source/composeK.js (hunk 2 above)
Comment (niksosf) [reply source/composeK.js]: Fair point. we could do something like this
```js
if (init.length === 0) {
    return last;
  }
  return compose(compose.apply(this, map(chain, init)), last);
```

**Comment by niksosf** [comment]
@CrossEye The previous tests must not have covered the case of only one function being passed to composeK, thus never reaching the body of the first if condition. maybe we should add a test for it. If agreed, I am up for it.

File: source/composeK.js [hunk 3]
```diff
@@ -2,7 +2,6 @@ import chain from './chain';
 import compose from './compose';
 import map from './map';
 
-
```
Comment (davidchambers) [inline source/composeK.js]: Please revert this change.

File: yarn.lock [hunk 4]
```diff
@@ -0,0 +1,3784 @@
+# THIS IS AN AUTOGENERATED FILE. DO NOT EDIT THIS FILE DIRECTLY.
```
Comment (davidchambers) [inline yarn.lock]: Please remove this file from the pull request.

File: source/composeK.js [hunk 5]
```diff
@@ -39,5 +38,8 @@ export default function composeK() {
   }
   var init = Array.prototype.slice.call(arguments);
   var last = init.pop();
+  if (init.length === 0) {
+    return last;
+  }
```
Comment (davidchambers) [inline source/composeK.js]: Even nicer, I think:

```javascript
return init.length === 0 ?
  last :
  compose(compose.apply(this, map(chain, init)), last);
```

Expressions are just about always preferable to statements. :)

File: source/composeK.js (hunk 3 above)
Comment (niksosf) [reply source/composeK.js]: Hi there, you mean the @@ -2,7 +2,6 @@ import chain from './chain'; part correct? I am looking at the source now and it is importing chain. If this is a git history thing, how does one remove this bit of change from the commit?

File: source/composeK.js (hunk 5 above)
Comment (niksosf) [reply source/composeK.js]: I was totally going to do that;) then I saw the if statement up top and resorted to conform to my imaginary standard that Ramda team might have had...

File: source/composeK.js (hunk 3 above)
Comment (niksosf) [reply source/composeK.js]: Ahh, it was prettier auto-formatting then eslint restoring the single-quotes that triggered that change. I am going to reset everything to master and reapply all the changes then .

**Comment by CrossEye** [comment]
Yes, I would very much like a test case that shows what this change supports that wasn't supported before.

**Comment by niksosf** [comment]
Sorry took so long, will write extra test cases for it now.
